### PR TITLE
Organize sensor fusion

### PR DESCRIPTION
**Pull request by antimonyB**
Moved Mahony files and refactored code.

### CODE REVIEW

File: Autopilot/Inc/gps.hpp
Comment (Pop0097) [inline Autopilot/Inc/gps.hpp]: pls fix to match style guide: https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/1827635203/Style+Guide

File: Autopilot/Src/gps.cpp
Comment (Pop0097) [inline Autopilot/Src/gps.cpp]: style guide: https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/1827635203/Style+Guide

`GpsData_t& Data`

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp:129]: There a reason we are remving the airspeed data? Or is it not needed anymore

**Review by Pop0097 — CHANGES_REQUESTED** [review]
Style guide stuff mostly

Would it also be possible to define the functions for getting sensor data for airspeed, etc? I don't think they've been done

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (antimonyB) [reply Autopilot/SensorFusion/Src/SensorFusion.cpp:129]: The original attitude function included airspeed but never actually used the data, so it's not needed.

Comment (Pop0097) [reply Autopilot/SensorFusion/Src/SensorFusion.cpp:129]: sounds good

**Review by Pop0097 — COMMENTED** [review]
Looks good to me. Just wanted to confirm how your SF tests performed locally and then I can approve

**Comment by antimonyB** [comment]
After these changes I don't see any issues with the SF tests.

**Comment by Pop0097** [comment]
Awesome! We will merge this in after the S2021 branch is merged in!

File: Autopilot/CMakeLists.txt
Comment (Pop0097) [inline Autopilot/CMakeLists.txt]: All of this (lines 251 to 255) should be in `SENSOR_FUSION_MODULES_SOURCES`, not `SENSOR_FUSION_MODULES_UNIT_TEST_SOURCES`

`SENSOR_FUSION_MODULES_UNIT_TEST_SOURCES` should only contain the university ttest files. 

Same applies for the attitude manager tests on line 173 and 174.

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: Could use memcpy here:

https://en.cppreference.com/w/cpp/string/byte/memcpy

Same applies to other Get_SensorData functions

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: https://en.cppreference.com/w/cpp/string/byte/memcpy

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: I'd prefer if you used `sizeof(STRUCT_YOU_ARE_COPYING)` here instead.

ex. `sizeof(IMU_Data_t)`

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: `sizeof(Airspeed_Data_t)`

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: `sizeof(Gps_Data_t)`

File: Autopilot/SensorFusion/Src/SensorFusion.cpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Src/SensorFusion.cpp]: `sizeof(Altimeter_Data_t)`

File: Autopilot/SensorFusion/Inc/SensorFusion.hpp
Comment (Pop0097) [inline Autopilot/SensorFusion/Inc/SensorFusion.hpp]: comment does not conform to datatype used. 

Also, the GPS struct uses int for heading. For consistency, we should use `int` here too

**Review by Pop0097 — CHANGES_REQUESTED** [review]
Small comments. Also, please address the changes required in the CMake file

**Review by Pop0097 — APPROVED** [review]
Looks good to me

**Pull request merged**
